### PR TITLE
Remove bitcast workaround and update `ExpOpConversionApprox`

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

**Comment by etiotto** [comment]
Rebase once PR #2981  lands

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 1]
```diff
@@ -1356,14 +1297,7 @@ struct AbsFOpConversion
                                    ConversionPatternRewriter &rewriter,
                                    Type elemTy, MultipleOperandsRange operands,
                                    Location loc) const {
-    // FIXME: Remove bitcast to and from i16 once SPIRV-LLVM-Translator supports
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1359]: Don't think SPIRV-LLVM-Translator added the support.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 2]
```diff
@@ -1373,8 +1307,8 @@ struct AbsFOpConversion
       auto maskAttr = rewriter.getIntegerAttr(elemTy, mask);
       auto maskConst = rewriter.create<LLVM::ConstantOp>(loc, maskAttr);
       Value res = and_(v, maskConst);
-      if (llvm::isa<BFloat16Type>(orig_type))
-        res = bitcast(res, orig_type);
+      if (llvm::isa<BFloat16Type>(elemTy))
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1310]: Assuming SPIRV-LLVM-Translator added the support, then we don't need to convert back, as we removed the original conversion at line 1363.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp (hunk 1 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1359]: Are we missing a test then ? Nothing fails.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp (hunk 2 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1310]: Right

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1359]: Let's wait, I think one UT would fail.

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1359]: `language/test_core.py::test_abs[bfloat16]` fails.

**Review by whitneywhtsang — APPROVED** [review]

**Comment by alexbaden** [comment]
Can you add a description?

**Comment by etiotto** [comment]
> Can you add a description?

I think the abstract is descriptive enough.

**Comment by etiotto** [comment]
> > Can you add a description?
> 
> I think the abstract is descriptive enough.

Actually the abstract is was stale. I have updated it to be more reflective of the change in the PR

**Pull request merged**